### PR TITLE
feature(testcafe): Search scenario

### PR DESCRIPTION
**Pull request by CozyKourai**

### CODE REVIEW

**Comment by CozyKourai** [comment]
[(𝐒) SCN - recherche](https://trello.com/c/f22W7VXM/1704-%F0%9D%90%92-scn-recherche)

File: testcafe/tests/drive/search.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import { driveUser } from '../helpers/roles'
+//import { Role } from 'testcafe'
```
Comment (y-lohse) [inline testcafe/tests/drive/search.js]: Please remove commented lines if they aren't used.

File: testcafe/tests/drive/search.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    text: 'a',
+    screenshotPath: `${FEATURE_PREFIX}/${TEST_SEARCH1}-A`
+  })
+
+  await privateDrivePage.typeInSearchInput({
+    text: 't',
+    screenshotPath: `${FEATURE_PREFIX}/${TEST_SEARCH1}-T`
+  })
+  await privateDrivePage.openSearchResultByIndex(0)
+  // For now, there is a bug when opening the viewer from search. So waitForLoading and screenshot are commented to avoid unstable tests
```
Comment (y-lohse) [inline testcafe/tests/drive/search.js]: Is that bug referenced somewhere?

File: testcafe/tests/pages/drive/drive-model-private.js [hunk 3]
```diff
@@ -460,4 +461,27 @@ export default class privateDrivePage extends DrivePage {
...
   }
+
+  async typeInSearchInput({
+    text: text,
+    screenshotPath: screenshotPath,
+    withMask = false
+  }) {
+    await isExistingAndVisibile(selectors.searchInput, 'selectors.searchInput')
+    await t.typeText(selectors.searchInput, text)
+    //wait 1s for result to appears
```
Comment (y-lohse) [inline testcafe/tests/pages/drive/drive-model-private.js]: I think it's 3s now, but maybe best not to put a number here?

File: testcafe/tests/pages/drive/drive-model-private.js [hunk 4]
```diff
@@ -460,4 +461,27 @@ export default class privateDrivePage extends DrivePage {
...
+    await t.wait(SEARCH_DELAY)
+    await t.fixtureCtx.vr.takeScreenshotAndUpload({
+      screenshotPath: `${screenshotPath}`,
```
Comment (y-lohse) [inline testcafe/tests/pages/drive/drive-model-private.js]:
```suggestion
      screenshotPath,
```

No need to interpolate the variable here.

File: testcafe/tests/drive/search.js (hunk 1 above)
Comment (CozyKourai) [reply testcafe/tests/drive/search.js]: I removed `//import { Role } from 'testcafe'` but left `//import ViewerDrivePage from '../pages/drive-viewer/drive-viewer-model'` and `//const viewerDrivePage = new ViewerDrivePage()` as theyĺl be needed when un-commenting the part about the viewer bug

**Comment by CozyKourai** [comment]
About screenshot in method, we actually have a few functions that does it, like `deleteCurrentFolder` (https://github.com/cozy/cozy-drive/blob/master/testcafe/tests/pages/drive/drive-model-private.js#L328-L330) but only because I want to screenshot the modal, so I need to take a screenshot "in the middle" of executing this method.

Here in `typeInSearchInput` where taken at the end of the method, so there's no need to keep it. I removed it and add the screenshots in the scenario

File: testcafe/tests/drive/search.js (hunk 2 above)
Comment (CozyKourai) [reply testcafe/tests/drive/search.js]: Thomas talk about it in the trello card for this PR. 
I reproduce it while testing this test. I'll add a trello link in code if there is a card for it (did find it for now, waiting for thomas to answer on mattermost! ;) )

Comment (CozyKourai) [reply testcafe/tests/drive/search.js]: https://trello.com/c/N8kFROf3

**Pull request merged**